### PR TITLE
Pauseable garrison

### PR DESCRIPTION
**Pull request by wolfbyte**
I'm not sure if we want to use dmg_crit as the condition, but it seemed the best option I had. 
Another option is to just create a new condition. 

You are the yaml master. This works though.

### CODE REVIEW

**Comment by MustaphaTR** [comment]
@GraionDilach It doesn't seem like changes removed your +1, but can you review again there were some changes.

I tested myself and it looks fine.

**Comment by GraionDilach** [comment]
I don't like this now - pausing has no effect. There is literal no reason to make this pausable atm, plain Conditionaltrait would be enough.

**Comment by MustaphaTR** [comment]
Ermm, pausing currently disable entering without forcing the units out.

**Comment by GraionDilach** [comment]
Oh. I see. That's still bugged though.

It's only checked during issuing the garrison order but it's not rechecked before entering in. EnterGarrsion.TryStartEnter was unupdated.

File: OpenRA.Mods.RA2/Activities/EnterGarrison.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ protected bool TryStartEnter(Actor self, Actor targetActor)
 
 			// Make sure we can still enter the transport
 			// (but not before, because this may stop the actor in the middle of nowhere)
-			if (enterGarrison == null || !garrisoner.Reserve(self, enterGarrison))
+			if (enterGarrison == null || !garrisoner.Reserve(self, enterGarrison) || enterGarrison.IsTraitDisabled || enterGarrison.IsTraitPaused)
```
Comment (GraionDilach) [inline OpenRA.Mods.RA2/Activities/EnterGarrison.cs]: I'd shift the reserving check to last one tbh since that's more expensive than the boolean lookups.

Comment (wolfbyte) [reply OpenRA.Mods.RA2/Activities/EnterGarrison.cs]: Done.

**Pull request merged**